include/util/json_deep_compare.hpp

```diff
@@ -0,0 +1,158 @@
+#ifndef UTIL_JSON_DEEP_COMPARE_HPP
+#define UTIL_JSON_DEEP_COMPARE_HPP
+
+#include "util/json_container.hpp"
+#include "util/integer_range.hpp"
+
+#include <boost/assert.hpp>
+
+#include <algorithm>
+#include <functional>
+#include <set>
+
+namespace osrm
+{
+namespace util
+{
+namespace json
+{
+
+struct Comparator : mapbox::util::static_visitor<bool>
+{
+    Comparator(std::string &reason_, const std::string &lhs_path_, const std::string &rhs_path_)
+        : reason(reason_), lhs_path(lhs_path_), rhs_path(rhs_path_)
+    {
+    }
+
+    bool operator()(const String &lhs, const String &rhs) const {
+      bool is_same = lhs.value == rhs.value;
+      if (!is_same)
+      {
+        reason = lhs_path + " (= \"" + lhs.value + "\") != " + rhs_path + " (= \"" + rhs.value + "\")";
+      }
+      return is_same;
+    }
+
+    bool operator()(const Number &lhs, const Number &rhs) const {
+      bool is_same = lhs.value == rhs.value;
```

Comment:
This compares all numbers for `==`, what if we want to test routes that e.g. do not return `0` as `duration` / `distance`? That is, we need a way of saying: compare the JSON but don't complain on e.g. on the `distance` field.

Or what is your take on this?